File: task7_1.js
// Read inputs from Standard Input (use readline()).
// Write outputs to Standard Output (use print()).

var n = parseInt(readline());
var data = readline().split(/\s+/);

var maxi = parseInt(data[0]);
var perte = 0;
for (var x = 1; x < n; x++) {
  if (maxi - parseInt(data[x]) > perte) {
    perte = maxi - parseInt(data[x]);
  }
  if (parseInt(data[x]) > maxi) {
    maxi = parseInt(data[x]);
  }
}
print(-perte);

File: task7_1.scala
// Stock Exchange Losses
// Read inputs from System.in, Write outputs to use print.
// Your class name has to be Solution
import math._
import scala.util._

object Solution {
  def main(args: Array[String]) {
    val n = readInt
    val data = readLine.split("\\s+");
    var maxi = data(0).toInt;
    var perte = 0;
    for(i <- 1 to n - 1) {
      if (maxi - data(i).toInt > perte) {
        perte = maxi - data(i).toInt;
      }
      if (data(i).toInt > maxi) {
        maxi = data(i).toInt;
      }
    }
    println(-perte);
  }
}
